flask_hw/flask_rest/api/room/rooms.py

```diff
@@ -0,0 +1,140 @@
+import logging
+
+from flask_restful import Resource, marshal_with, marshal
+from sqlalchemy.exc import SQLAlchemyError
+
+from api.room.room_parsers import data_valid_for
+from api.room.structure import room_structure
+from db import Rooms, db
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class RoomsRes(Resource):
+
+    @marshal_with(room_structure)
+    def get(self, number=None):
+        room_all = Rooms.query.all()
+        header = {
+            "Cache-Control": "no-store, no-cache, must-revalidate",
+            "Pragma": "no-cache"
+        }
+        args = data_valid_for('GET')
+        logging.debug(room_all)
+        logging.debug(number)
+        logging.debug(args)
+        if number:
+            room = Rooms.query.get(number)
+            if room:
+                return room, header
+            else:
+                # FIXME: how to display message instead of Rooms
+                #  structure with nulls?
+                return {"message": "room not found"}, 404, header
```

Comment:
it will be fine

Reply:
also I do not think you need to return header

Reply:
If an object with such id does not exist, the server returns `404 Not Found`
The response must include headers regarding the caching policy, as browsers actively cache GET and HEAD requests. In the absence of any cache management policy, there should be:
```
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
```
Principles of building a REST JSON API
https://m.habr.com/ru/post/447322/